notebook/static/base/js/utils.js

```diff
@@ -10,34 +10,45 @@ define([
 ], function($, CodeMirror, moment){
     "use strict";
     
+    /**
+     * Load a single extension.
+     * @param  {string} extension - extension path.
+     * @return {Promise} that resolves to an extension module handle
+     */
+    var load_extension = function (extension) {
+        return new Promise(function(resolve, reject) {
+            require(["nbextensions/" + extension], function(module) {
+                console.log("Loaded extension: " + extension);
+                try {
+                    module.load_ipython_extension();
+                } finally {
+                    resolve(module);
+                }
+            }, function(err) {
+                reject(err);
+            });
+        });
+    };
+
+    /**
+     * Load multiple extensions.
+     * Takes n-args, where each arg is a string path to the extension.
+     * @return {Promise} that resolves to a list of loaded module handles.
+     */
     var load_extensions = function () {
         // load one or more Jupyter notebook extensions with requirejs
-        
+
         var extensions = [];
         var extension_names = arguments;
         for (var i = 0; i < extension_names.length; i++) {
-            extensions.push("nbextensions/" + arguments[i]);
+            extensions.push(arguments[i]);
         }
```

Comment:
This loop seems a bit ridiculous now, but IIRC, `arguments` is an array like object rather than a real array, so it may be lacking methods like `.map()`. Is there a better way to get from array-like to array than looping over it?